avr/platform.txt

```diff
@@ -64,8 +65,28 @@ recipe.S.o.pattern="{compiler.path}{compiler.c.cmd}" {compiler.S.flags} -mmcu={b
 archive_file_path={build.path}/{archive_file}
 recipe.ar.pattern="{compiler.path}{compiler.ar.cmd}" {compiler.ar.flags} {compiler.ar.extra_flags} "{archive_file_path}" "{object_file}"
 
-## Combine gc-sections, archives, and objects
-recipe.c.combine.pattern="{compiler.path}{compiler.c.elf.cmd}" {compiler.c.elf.flags} -mmcu={build.mcu} {compiler.c.elf.extra_flags} -o "{build.path}/{build.project_name}.elf" {object_files} "{build.path}/{archive_file}" "-L{build.path}" -lm
+################################################################################
+## PLEASE NOTE: The way we link our elf binary significantly differs from the 
+##              way that Arduino handles things by default. 
+##              The original linker command has been split up into two 
+##              steps (recipe.c.combine.pattern and recipe.hooks.linking.postlink.1.pattern).
+##              This is necessary to prevent link errors reporting unresolved symbols
+##              due to order dependencies of libraries and objects appearing
+##              in the linker command line.
+##              
+## NEWLY INTRODUCED: Use java to delete the joined archive file. 
+## This is required as Arduino does not support any scripting features. 
+## Therefore, there's no direct command to erase a file.
+## The java code of class DeleteFile as well as the compiled byte code
+## can be found in the build-tools directory of this repo.
+recipe.hooks.linking.prelink.1.pattern="{runtime.ide.path}/java/bin/java" -cp "{runtime.platform.path}/../build-tools/auxiliary" DeleteFile "{build.path}/{build.project_name}_joined.a"
```

Comment:
Unfortunately, the Arduino package for macOS doesn't seem to include an independent java interpreter, so this is not a universal path to a `java` executable.